pkg/tools/prefix_pool/prefixpool.go

```diff
@@ -0,0 +1,642 @@
+package prefix_pool
+
+import (
+	"math/big"
+	"net"
+	"sort"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+)
+
+/**
+All in one interface for managing prefixes.
+*/
+//type PrefixPool interface {
+//	/*
+//		Process ExtraPrefixesRequest and provide a list of prefixes for clients to use.
+//	*/
+//	Extract(connectionId string, family networkservice.IpFamily_Family, requests ...*networkservice.ExtraPrefixRequest) (srcIP *net.IPNet, dstIP *net.IPNet, requested []string, err error)
+//	Release(connectionId string) error
+//	GetConnectionInformation(connectionId string) (string, []string, error)
+//	GetPrefixes() []string
+//	Intersect(prefix string) (bool, error)
+//	ExcludePrefixes(excludedPrefixes []string) ([]string, error)
+//	ReleaseExcludedPrefixes(excludedPrefixes []string) error
+//}
+
+type prefixPool struct {
+	sync.RWMutex
+
+	basePrefixes []string // Just to know where we start from
+	prefixes     []string
+	connections  map[string]*connectionRecord
+}
+
+func (impl *prefixPool) GetPrefixes() []string {
+	return impl.prefixes
+}
+
+type connectionRecord struct {
+	ipNet    *net.IPNet
+	prefixes []string
+}
+
+func NewPrefixPool(prefixes ...string) (*prefixPool, error) {
+	//TODO: Add validation of input prefixes.
```

Comment:
I think we need to fix this TODO comment

Reply:
What kind of validation is the best? Just ensure that prefixes match certain pattern?

Reply:
I think we need to verify that each prefix can be represented as CIDR